crates/transaction-generator-lib/src/lib.rs

```diff
@@ -100,7 +100,7 @@ pub struct CounterState {
 }
 
 #[async_trait]
-pub trait TransactionExecutor: Sync + Send {
+pub trait ReliableTransactionSubmitter: Sync + Send {
```

Comment:
Not sure if I like this naming - calling this `Reliable` doesn't add much value since we don't have any `Unreliable` alternative. How about just `TransactionSubmitter` ?

Reply:
+1

Reply:
We do have unreliable client - that is raw RestClient, and submit_worker sends transactions being fine with transactions failing. 

This one is adding expectations to the API that it will wait for transactions to be committed successfully,  and fail otherwise.  It does retries,  resubmission of transactions etc. Check RestReliabkeTransactionSubmitter implementation.


This is used exclusively for initialization before transaction generation for the given workload starts.

Reply:
May be `RetryingTransactionSubmitter` is a better name in that case?

Reply:
I prefer Reliable, as retrying doesn't confer the meaning.

initialization needs to succeed, and so every single transaction that initialization creates needs to be executed successfully